window.c

```diff
@@ -70,6 +80,18 @@ void initWindow(App *app, Config *cfg)
   }
   app->window = SDL_CreateWindowFrom((void *)rootWindow);
   if (app->window == NULL) lwpLog(LOG_ERROR, "%s", SDL_GetError());
+}
+#endif
 
+void initWindow(App *app, Config *cfg)
+{
+#ifdef __WIN32
+  initWindowWin32(app, cfg);
+#elif __DARWIN
+  initWindowMacOS(app, cfg);
+#elif __LINUX
+  initWindowX11(app, cfg);
+#else
+#error "Unsupported platform"
```

Comment:
This error isn't going to be thrown, because `__LINUX` is defined in `platform_guard.h` by default. I suggest removing `#define __LINUX` from `platform_guard.h` and putting this error there instead

Reply:
Yeah, I honestly wasn't sure how to handle this properly. When relying on editor code completion (like VSCode's C/C++ extension), defines like that have to be configured in the extension, so the editor knows which code to compile. I'm defaulting to `__LINUX` because that way, the editor should resolve this by itself and the development experience isn't impaired for people working on Linux.

However, it is definitely a weird way to solve this, so I'd say you're right, we should just do it properly and error in the case that platform defines are missing.